client/assets/js/meter.js

```diff
@@ -154,56 +153,55 @@ function flipSwitch(){
 //++
 //++
 function knobRange(elem){
+  debugger;
     switch ($(elem).attr('class')) {
-      case "range-indicator long":
+      case "close-range-knob":
          if(deviceOn){
             $('.knob-light').removeClass('selected');
-            $('.knob-light', elem).addClass('selected');
+            $(".long .knob-light").addClass('selected');
           }
-          $('#knob>#knobImg').removeClass();
-          $('#knob>#knobImg').addClass('long-range-knob');
-          knobMode = 'long';
-          handleMeter();
+          $(elem).removeClass("close-range-knob");
+          $(elem).addClass("long-range-knob");
+          handleMeter('long');
           break;
-      case "range-indicator mid":
+      case "long-range-knob":
           if(deviceOn){
               $('.knob-light').removeClass('selected');
-              $('.knob-light', elem).addClass('selected');
+              $(".mid .knob-light").addClass('selected');
             }
-          $('#knob>#knobImg').removeClass();
-          knobMode = 'med';
-          handleMeter();
+          $('#knobImg').removeClass("long-range-knob");
+          $('#knobImg').addClass("mid-range-knob");
+          handleMeter('med');
           break;
-      case "range-indicator close":
+      case "mid-range-knob":
           if(deviceOn){
             $('.knob-light').removeClass('selected');
-            $('.knob-light', elem).addClass('selected');
+            $(".close .knob-light").addClass('selected');
           }
-          $('#knob>#knobImg').removeClass();
-          $('#knob>#knobImg').addClass('close-range-knob');
-          knobMode = 'short';
-          handleMeter();
+          $(elem).removeClass("mid-range-knob");
+          $(elem).addClass("close-range-knob");
+          handleMeter('short');
           break;
     }
 }
 //++
 //++
-function handleMeter(){
-    if (knobMode === 'long'){
+function handleMeter(range){
```

Comment:
This wont work as currently coded. The success sub-function of the getLocation function calls handleMeter. At that point in the code, it doesn't know what the mode is and was relying on the global variable.

It is fine if that gets shifted to passing a paramater, but the issue needs to be resolved before we can merge and I am not sure how to get that info to getLocation. My recommendation is to, at least for now, keep the global. It is simpler that way.